toolchain/check/return.h

```diff
@@ -0,0 +1,37 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_RETURN_H_
```

Comment:
I understand the desire for helper files like this, but maybe it should be named something like "return_and_returned_helpers" to make clearer the contents? "return" is a very overloaded term, so having a file with just that as the name might carry over that ambiguity, especially since it isn't specific to `return`.

Reply:
I prefer it the way it is. "_helpers" tells me nothing, so I'd not want to add that part. I would be OK with `return_and_returned`, and will switch to that if you have a strong preference here, but I think they're closely-enough connected that simply `return` is sufficient.

Reply:
My concern is that there are all the `handle_*` files, and by being named after a statement, I'm slightly disoriented that this isn't another `handle_*` file that handles that statement; that is, `handle_return` would be sensible to see. The same would be true of `return_and_returned`, so it wouldn't offer clarity on this point.

On a related note, `convert` doesn't go the same way for me since there's no `convert` statement, though it's still a little weird of a name to me.

Not a strong opinion that some additional disambiguation should be blocking, it's more that I was hoping to reduce the source of this confusion somehow.